Orange/classification/naive_bayes.py

```diff
@@ -48,22 +49,62 @@ def __init__(self, log_cont_prob, class_prob, domain):
 
     def predict_storage(self, data):
         if isinstance(data, Instance):
-            data = [data]
-        if len(data.domain.attributes) == 0:
-            probs = np.tile(self.class_prob, (len(data), 1))
+            data = np.atleast_2d(data.X)
+        elif isinstance(data, Storage):
+            data = data.X
```

Comment:
The `Storage` class does not have `X` - `Table` has it, and even there it is not declared properly :)
Joking aside, our Learners and Models can implement fit/predict which work with numpy arrays or fit_storage/predict_storage which do not assume/require numpy data.
Instead of `data = data.X` you could replace the implementation of predict_storage with the `predict` method. That can have implications for non-numpy based storages (SqlTable)...
In short - this change needs some consideration and should probably be in its own PR.

Reply:
Moved to #3540.